sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/polling/OperationLocationPollingStrategy.java

```diff
@@ -0,0 +1,142 @@
+ *           kept
+ */
+public class OperationLocationPollingStrategy<T, U> extends OperationResourcePollingStrategy<T, U> {
+
+    private static final ClientLogger LOGGER = new ClientLogger(OperationLocationPollingStrategy.class);
+
+    private static final HttpHeaderName OPERATION_LOCATION_HEADER
+        = HttpHeaderName.fromString("Operation-Location");
+
+    private static final TypeReference<PostPollResult> POST_POLL_RESULT_TYPE_REFERENCE
+        = TypeReference.createInstance(PostPollResult.class);
+
+    private final ObjectSerializer serializer;
+    private final String endpoint;
+
+    /**
+     * Creates an instance of the operation resource polling strategy.
+     *
+     * @param pollingStrategyOptions options to configure this polling strategy.
+     * @throws NullPointerException if {@code pollingStrategyOptions} is null.
+     */
+    public OperationLocationPollingStrategy(PollingStrategyOptions pollingStrategyOptions) {
+        super(OPERATION_LOCATION_HEADER, pollingStrategyOptions);
+        this.endpoint = pollingStrategyOptions.getEndpoint();
+        this.serializer = pollingStrategyOptions.getSerializer() != null
+            ? pollingStrategyOptions.getSerializer() : JsonSerializerProviders.createInstance(true);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<PollResponse<T>> onInitialResponse(Response<?> response, PollingContext<T> pollingContext,
+                                                   TypeReference<T> pollResponseType) {
+        HttpHeader operationLocationHeader = response.getHeaders().get(OPERATION_LOCATION_HEADER);
+        if (operationLocationHeader != null) {
+            pollingContext.setData(OPERATION_LOCATION_HEADER.getCaseSensitiveName(),
+                PollingUtils.getAbsolutePath(operationLocationHeader.getValue(), endpoint, LOGGER));
+        }
+        final String httpMethod = response.getRequest().getHttpMethod().name();
+        pollingContext.setData(PollingConstants.HTTP_METHOD, httpMethod);
+
+        if (response.getStatusCode() == 200
+            || response.getStatusCode() == 201
+            || response.getStatusCode() == 202
+            || response.getStatusCode() == 204) {
+            final Duration retryAfter = ImplUtils.getRetryAfterFromHeaders(response.getHeaders(), OffsetDateTime::now);
+            final Mono<PollResponse<T>> pollResponseMono;
+            if (HttpMethod.PUT.name().equalsIgnoreCase(httpMethod)
+                || HttpMethod.PATCH.name().equalsIgnoreCase(httpMethod)) {
+                // PUT has initial response body as resultType
+                // we expect Response<?> be either Response<BinaryData> or Response<U>
+                // if it is not Response<BinaryData>, PollingUtils.serializeResponse would miss read-only properties
+                pollResponseMono = PollingUtils.serializeResponse(response.getValue(), serializer)
```

Comment:
The `PollingUtils.serializeResponse` could have problem when `Response<U>`. Details at issue https://github.com/Azure/azure-sdk-for-java/issues/34248 (PR to demonstrate https://github.com/Azure/azure-sdk-for-java/pull/34247)

Previously, this may not be a big problem, as the response of activation operation would only be a PollResult, which even if not converted correctly (e.g. "status" property missing), could still be safely regarded as `IN_PROGRESS`.

However, when this response of activation operation would be used as final result, any error could be serious problem.

Here is a prototype of modified JsonSerializer that can mitigate the problem, but it may not be a good solution.

Customized JacksonAdapter
https://github.com/Azure/azure-sdk-for-java/blob/58324d969f9ead60ad8255418accb9896b4ba6bf/sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollingUtilsTests.java#L104-L120
---

Currently codegen would always use `Response<BinaryData>`, but this is only a happy coincidence.